### ISSUE TITLE
Fresh Install Failing on a Langchain Module

### ISSUE THREAD

**Issue by sergeyktvg** [gpt-engineer-org/gpt-engineer#921]
Hi there, first time user, so apologies if I'm just being dumb and didn't understand the instructions...

However, this is the issue I'm seeing when I try to run gpt-engineer:
```ImportError: cannot import name 'get_openai_token_cost_for_model' from 'langchain.callbacks.openai_info'```

I have tried uninstalling and reinstalling langchain. The environment variable for my openai api key is set. Sorry again for confusion, just need some guidance if someone can advise.

**Comment by ruedaminute** [comment]
seeing the same problem also!

**Comment by ghovax** [comment]
Same here, same issue.

**Comment by nrsmac** [comment]
I did @ghovax 's fix but then received another import error.  
```
Traceback (most recent call last):
  File "/Users/nrsmac/repos/inspira/venv/bin/gpt-engineer", line 5, in <module>
    from gpt_engineer.applications.cli.main import app
  File "/Users/nrsmac/repos/inspira/venv/lib/python3.11/site-packages/gpt_engineer/applications/cli/main.py", line 40, in <module>
    from gpt_engineer.applications.cli.file_selector import ask_for_files
  File "/Users/nrsmac/repos/inspira/venv/lib/python3.11/site-packages/gpt_engineer/applications/cli/file_selector.py", line 44, in <module>
    import tkinter as tk
  File "/opt/homebrew/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/tkinter/__init__.py", line 38, in <module>
    import _tkinter # If this fails your Python may not be configured for Tk
```

**Comment by ErikBjare** [comment]
This happens because our version constraint for langchain is too relaxed.

When installed with poetry (as in CI), it works fine because the langchain version is pinned. But since the version requirement is set as `>=0.0.350` anyone installing with pip would get a newer version on install, which has breaking changes. 

We will restrict the langchain requirement to a specific version to avoid this in the future.

**Comment by gummipunkt** [comment]
> Same here, same issue. I fixed it with `pip3 install langchain==0.0.350`.

works for me too, had also to install tkinter
brew install python-tk (on macos)
